spring-statemachine-core/src/main/java/org/springframework/statemachine/transition/AbstractTransition.java

```diff
@@ -87,6 +104,7 @@ public AbstractTransition(State<S, E> source, State<S, E> target, Collection<Act
 		this.guard = guard;
 		this.trigger = trigger;
 		this.securityRule = securityRule;
+		this.name = name;
```

Comment:
You could do null check here as set it to empty, then from other constructors pass in null.